src/System.Runtime.Extensions/src/System.Runtime.Extensions.csproj

```diff
@@ -176,6 +176,35 @@
       <Link>Common\Interop\Windows\mincore\Interop.LookupAccountNameW.cs</Link>
     </Compile>
   </ItemGroup>
+  <ItemGroup Condition=" '$(TargetsWindows)' == 'true' And '$(CoreClrOrCorRt)' == 'true'"> <!-- Microsoft.Win32.Registry, needed for some Environment env var support -->
+    <Compile Include="..\..\Microsoft.Win32.Registry\src\Microsoft\Win32\Registry.cs" />
```

Comment:
Yes, it increases the size of the assembly from ~90kb to ~110kb.

I listed the options I'm aware of for supporting this.  They all have downsides:
1. We can expose from the runtime the implementation to use, but a) that exposes unnecessary support from the runtime, b) we'd only be exposing it from the runtime on some platforms, and c) it's not necessarily in all runtimes we want this to work on, e.g. in the future when using corert in an environment that supports this.
2. We can use Microsoft.Win32.Registry.dll via reflection, but that will then fail if that assembly isn't available when the app runs.
3. We can write a custom implementation that includes the bare minimum of code, but as this is both reading and writing various registry keys, it pulls in a fair part of the implementation already.  Just from the one test I added, a code coverage run shows ~30% of the imported code getting hit, and that doesn't include error paths and the like that are possible.
4. We can simply compile in the code as I've done, which increases the size of the assembly by more than is strictly necessary.  As I noted in the PR description, we could subsequently choose to do some refactoring/restructuring on the implementation to try to pull in less, if we really think it's worthwhile.

If you prefer one of the other options or have suggestions for others, I'm open to them.  But we're exposing an API here that is fundamentally a registry-access API... even the MSDN docs for it describe it as such: "Retrieves the value of an environment variable from the current process or from the Windows operating system registry key for the current user or local machine."